awx/main/tasks.py

```diff
@@ -1388,6 +1393,10 @@ def run(self, pk, **kwargs):
         if self.instance.spawned_by_workflow:
             self.parent_workflow_job_id = self.instance.get_workflow_job().id
 
+        # TODO: can we count on instance always having created?
```

Comment:
Yes.  If we get to this point where we're creating an actual event, the related job already exists in the database, and definitely has a `created` value.

Reply:
Also, this column is not nullable:

```sql
                                             Table "public.main_unifiedjob"
         Column          |           Type           | Collation | Nullable |                   Default
-------------------------+--------------------------+-----------+----------+---------------------------------------------
 id                      | integer                  |           | not null | nextval('main_unifiedjob_id_seq'::regclass)
 polymorphic_ctype_id    | integer                  |           |          |
 created                 | timestamp with time zone |           | not null |
```

Reply:
The db as the single source of truth on existence of data - that's cool / makes sense 👍 

Also, makes sense that the job _better_ have a created value. It was just that as I was going through keying off of things I wondered about taking for granted the existence of fields. Good to know ^